### ISSUE TITLE
INTERNAL Error: Invalid unicode (byte sequence mismatch) detected in segment statistics update

### ISSUE THREAD

**Issue by manticore-projects** [duckdb/duckdb#7263]
### What happens?

```text
Appender10MRows	1	2.634649
Appender10MRows	2	2.635860
Appender10MRows	3	2.641729
Appender10MRows	4	2.636160
Appender10MRows	5	2.630352
Appender10MRowsPrimaryKey	1	8.375196
Appender10MRowsPrimaryKey	2	terminate called after throwing an instance of 'duckdb::InternalException'
  what():  INTERNAL Error: Invalid unicode (byte sequence mismatch) detected in segment statistics update
```

### To Reproduce

Normal Performance Test, called by
```shell
build/release/benchmark/benchmark_runner
```

### OS:

Linux archlinux 6.2.12-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 20 Apr 2023 16:11:55 +0000 x86_64 GNU/Linux

### DuckDB Version:

Latest Git

### DuckDB Client:

Shell

### Full Name:

Andreas Reichel

### Affiliation:

manticore-projects Co. Ltd.

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by manticore-projects** [comment]
Yes, that was my assumption too and so I reported this issue.
However, I can look into this only much later, after fixing some JDBC stuff. We can keep it as a reminder. Can we assign priorities?

**Comment by lnkuiper** [comment]
I have a hunch about what it could be.

We could try running the benchmark with the build flag `DESTROY_UNPINNED_BLOCKS=1` and `DISABLE_STRING_INLINE=1` and see if we can reproduce.

**Comment by rsund** [comment]
Reproducible parquet in #5882 (although not benchmark related).

**Comment by Tishj** [comment]
I'm gonna take a guess at the issue, from looking into it a bit.
TL;DR the pointer is saved directly and it outlives the scope

We create a stack allocated `char area_code[6];`, then we create a string_t from this memory
if `DUCKDB_DEBUG_NO_INLINE` is set, we store this pointer directly.
at:
```c++
string_t StringVector::AddStringOrBlob(Vector &vector, string_t data) {
	D_ASSERT(vector.GetType().InternalType() == PhysicalType::VARCHAR);
	if (data.IsInlined()) {
		// string will be inlined: no need to store in string heap
		return data;
	}
	if (!vector.auxiliary) {
		vector.auxiliary = make_buffer<VectorStringBuffer>();
	}
	D_ASSERT(vector.auxiliary->GetBufferType() == VectorBufferType::STRING_BUFFER);
	auto &string_buffer = (VectorStringBuffer &)*vector.auxiliary;
	return string_buffer.AddBlob(data);
}
```
This pointer is also not allocated, because it's supposed to be inlined, so we store the pointer in the Vector.
